### PR TITLE
[New Template] operations/terraform-state.yaml

### PR DESCRIPTION
**Pull request by andreaswittig**
Create S3 bucket and DynamoDB table to manage Terraform remote state. Use bucket policy to avoid leaking sensitive information stored in remote state. Also, encrypt sensitive data.

### CODE REVIEW

File: operations/terraform-state.yaml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    Type: AWS::DynamoDB::Table
+    Properties:
+      AttributeDefinitions:
+      - AttributeName: 'LockID'
+        AttributeType: 'S'
+      BillingMode: 'PAY_PER_REQUEST'
+      KeySchema:
+      - AttributeName: 'LockID'
+        KeyType: HASH
+      SSESpecification:
```
Comment (michaelwittig) [inline operations/terraform-state.yaml:93]: we could use the key from ParentKmsKeyStack?

File: operations/terraform-state.yaml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+          Action: 's3:*'
+          Resource:
+          - !Sub '${StateBucket.Arn}'
+          - !Sub '${StateBucket.Arn}/*'
+          NotPrincipal:
+            AWS: !Split [',', {"Fn::Sub": '${TerraformStateUserARNs},${TerraformStateAdminARNs}'}]
+        - Effect: 'Deny'
+          NotAction:
+          - 's3:ListBucket'
+          - 's3:GetObject'
```
Comment (michaelwittig) [inline operations/terraform-state.yaml:76]: what about

* `s3:ListBucketVersions`
* `s3:GetObjectVersion`

File: operations/terraform-state.yaml (hunk 1 above)
Comment (andreaswittig) [reply operations/terraform-state.yaml:104]: The DynamoDB table does not contain sensitive information.

File: operations/terraform-state.yaml (hunk 2 above)
Comment (andreaswittig) [reply operations/terraform-state.yaml]: Terraform does not need access to those actions.

Comment (michaelwittig) [reply operations/terraform-state.yaml]: I see

File: operations/terraform-state.yaml (hunk 1 above)
Comment (michaelwittig) [reply operations/terraform-state.yaml:104]: I see

**Pull request merged**